Finished homework

jfredz opened the pull request:
Please inspect my homework

gregurco commented on src/components/Questions/List.vue:
```diff
@@ -0,0 +1,84 @@
+      />
+    </div>
+
+    <div v-for="question in questions" v-else :key="question.id">
```
it's not a good practice to keep v-for and v-else in same element. Split them in next mode:
```
<template v-else>
    <div v-for="question in questions" :key="question.id">
         // ...
    </div>
</template>
```

jfredz replied:
Removed v-else from the div, and nested everything in <template> with the v-else

gregurco commented on src/views/User.vue:
```diff
@@ -0,0 +1,51 @@
+      fetchUser: 'profile/fetch'
+    })
+  },
+  mounted() {
```
fetch users in `created` event, not `mounted`

jfredz replied:
Moved the fetch to the created event instead of mounted

gregurco commented on src/views/Tag.vue:
```diff
@@ -0,0 +1,49 @@
+import { mapGetters, mapActions } from 'vuex'
+
+export default {
+  props: {
```
add `name` to component

jfredz replied:
added name to the component

gregurco commented on src/components/Questions/List.vue:
```diff
@@ -0,0 +1,84 @@
+<script>
+import { mapActions, mapGetters } from 'vuex'
+export default {
+  name: 'Home',
```
wrong name of component

jfredz replied:
changed the name of the component

gregurco commented:
Too many problems with code style. Please run eslint and fix all problems.

jfredz commented:
I've run eslint and fixed the issues. So sorry about that. 
However, I have a question about an eslint warning. Basically, I've used the `v-html` directive in order to display correct names and locations for users. Sure thing I was getting warnings from eslint, and I just added this comment to remove them:
`<!-- eslint-disable-next-line vue/no-v-html -->`

Is it allowed to do so? 

And thank you for the review

gregurco commented:
@jfredz yes, it's ok to use `v-html` in case you know the source of data and you trust it (data sanitized from backend). So, no problem with usage of `v-html`.

gregurco commented on src/components/user/LastUserWidget.vue:
```diff
@@ -0,0 +1,56 @@
+        indeterminate
+      />
+    </div>
+    <div v-for="user in userList" v-else :key="user.id" class="avatars">
```
Please fix problem with `v-else` here too

jfredz replied:
Wrapped the v-else div in a template, as well as the loader div, so that they are wrapped in the same tag

gregurco commented on src/components/user/LastUserWidget.vue:
```diff
@@ -0,0 +1,56 @@
+      loading: 'user/getLoading'
+    })
+  },
+  mounted () {
```
Also use `created` here and check if there are same cases.

jfredz replied:
Changed to `created`

gregurco commented on src/router/router.js:
```diff
@@ -0,0 +1,60 @@
+        {
+          path: 'question/:id',
+          name: 'question',
+          'params': ['id'],
```
why param is as string and not just `params: ['id'],` ?

jfredz replied:
Indeed, I just noticed that. Makes sense because it's just an object key.  Changed to `params`

jfredz commented:
Fixed all issues. As far as I understand, we have to make fetch from API backend at `created` instead of `mounted` because created is called earlier and basically all API requests and asynchronous tasks should be at `created`.

gregurco commented:
> Fixed all issues. As far as I understand, we have to make fetch from API backend at `created` instead of `mounted` because created is called earlier and basically all API requests and asynchronous tasks should be at `created`.

yes, you're right

gregurco commented:
@rapi could you please also review?

rapi commented on src/components/Search/Search.vue:
```diff
@@ -0,0 +1,88 @@
+}
+</script>
+
+<style scoped>
```
Will be great if you will put it in different component

rapi replied:
styles

rapi commented on src/components/Search/Search.vue:
```diff
@@ -0,0 +1,88 @@
+  name: 'Search',
+  computed: {
+    ...mapGetters({
+      usersList: 'search/getUsersList',
```
Loading

rapi commented on src/components/layouts/CommonLayout.vue:
```diff
@@ -0,0 +1,35 @@
+
+export default {
+  name: 'CommonLayout',
+  components: { LeftMenu, Header, RightPart, Search }
```
Lint it

jfredz replied:
Done

rapi commented on src/components/layouts/Header.vue:
```diff
@@ -0,0 +1,64 @@
+      fetchTags: 'search/mutateTagsList'
+    }),
+    fetchQueries () {
+      this.fetchUsers(this.getSearchPanelValue)
```
loading

rapi commented on src/components/layouts/LeftMenu.vue:
```diff
@@ -0,0 +1,36 @@
+<template xmlns:v-slot="http://www.w3.org/1999/XSL/Transform">
```
can i know for what you need  xmlns:v-slot="http://www.w3.org/1999/XSL/Transform"?

jfredz replied:
There is no need for it here indeed. I was playing around with custom components to make the menu layout and somehow got it too. Removed it

rapi commented on src/components/layouts/LeftMenu.vue:
```diff
@@ -0,0 +1,36 @@
+      <v-list-tile>
+        <v-list-tile-content>
+          <v-list-tile-title>
+            <router-link :to="{name: 'home'}">Home</router-link>
```
better to put in array

rapi commented on src/components/user/LastUserWidget.vue:
```diff
@@ -0,0 +1,60 @@
+    />
+    <template v-if="loading">
+      <div class="text-xs-center mt-5">
+        <v-progress-circular
```
Will be great if it will be in different component

jfredz replied:
Meaning the loader should be an independent component?

rapi commented on src/store/modules/user.js:
```diff
@@ -0,0 +1,46 @@
+  },
+  getters: {
+    getAllList: (state) => state.allList,
+    getLoadingAll: (state) => state.loadingAll,
```
getLoadingAll and getLoading you can use in same place

rapi replied:
you dont need to duplicate it

jfredz replied:
Makes a lot of sense, thanks. Removed `getLoadingAll`, and left only `getLoading`

rapi commented on src/store/modules/profile.js:
```diff
@@ -0,0 +1,22 @@
+import { fetchUsers } from '@/api/users'
```
You can merge it with Users module

jfredz replied:
I was thinking about this but wasn't sure. Merged it with user module, and removed the profile module

rapi commented on src/store/modules/search.js:
```diff
@@ -0,0 +1,65 @@
+import { fetchUsers } from '@/api/users'
```
You dont need this module, you can use users,questions,tags directly

jfredz replied:
Implemented this suggestion. Removed the search module, and used users, questions and tags modules. However, I decided to make separate lists for search fetches, because I don't want to mutate the same list. 

For example, If I am on users page, and make a search, If I use the same list in the module, I will mutate it with the search results and therefore mutate the user's page content. If I click on a user, there is no issue because I leave the users page, but what if I just change my mind and delete the search query? The search box should hide and I should see all users on the user's page again. 

Hope this is not too confusing, and please advise if this is the right approach!
Thank you
